Source/WarpX.cpp

```diff
@@ -1209,6 +1209,14 @@ WarpX::ReadParameters ()
         J_in_time = GetAlgorithmInteger(pp_psatd, "J_in_time");
         rho_in_time = GetAlgorithmInteger(pp_psatd, "rho_in_time");
 
+        if (psatd_solution_type != PSATDSolutionType::FirstOrder || do_multi_J == false)
+        {
+            WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+                rho_in_time == RhoInTime::Linear,
+                "psatd.rho_in_time=constant not yet implemented, "
+                "except for psatd.solution_type=first-order and warpx.do_multi_J=1");
+        }
+
```

Comment:
The function `PushPSATD`, modified above in Source/FieldSolver/WarpXPushFieldsEM.cpp, is used only without the multi-J algorithm, which means that we have only the spectral indices `rho_old` and `rho_new` since rho constant in time is not implemented without the multi-J algorithm, as this statement asserts.